tests/deprecated_api/test_remove_1-8.py

```diff
@@ -925,3 +925,24 @@ def test_root_gpu_property_0_passing(monkeypatch, gpus, expected_root_gpu, strat
         "Please use `Trainer.strategy.root_device.index` instead."
     ):
         assert Trainer(gpus=gpus, strategy=strategy).root_gpu == expected_root_gpu
+
+
+@pytest.mark.parametrize(
+    ["trainer_kwargs", "expected_num_processes"],
+    [
+        ({}, 1),
+        ({"devices": 1}, 1),
+        ({"devices": 4}, 4),
+        ({"accelerator": "cpu", "devices": 1}, 0),
+        ({"accelerator": "gpu", "devices": 4}, 4),
+    ],
+)
+def test_trainer_num_processes(monkeypatch, trainer_kwargs, expected_num_processes):
+    monkeypatch.setattr(torch.cuda, "is_available", lambda: True)
+    monkeypatch.setattr(torch.cuda, "device_count", lambda: 16)
```

Comment:
Only do the monkey patch when accelerator is `gpu`

Reply:
Updated, thanks!